test/services/human_time_service_test.dart

```diff
@@ -0,0 +1,116 @@
+import "package:flutter_test/flutter_test.dart";
+import "package:proxima/services/human_time_service.dart";
+
+import "../mocks/providers/provider_human_time_service.dart";
+
+void main() {
+  group("Human Time Service Unit tests", () {
+    late HumanTimeService humanTimeService;
+
+    setUp(() {
+      // Setup the `HumanTimeService` for constant time
+      humanTimeService = HumanTimeService(
+        currentDateTimeCallback: constantTimeCallback(),
+      );
+    });
+
+    test("Check correct simple absolute time", () {
+      // Setup actual values for simple date
+      final time1 = DateTime.utc(2002);
+
+      const expectedSimpleHumanText = "Tuesday, January 1, 2002 00:00";
+
+      // Use the service to get human time
+      final actualSimpleHumanText = humanTimeService.textTimeAbsolute(
+        time1,
+      );
+
+      // Check that the absolut time value is correct
+      expect(
+        actualSimpleHumanText,
+        expectedSimpleHumanText,
+      );
+    });
+
+    test("Check correct complex absolute time", () {
+      // Setup actual values for complex specific date
+      final time2 = DateTime.utc(
+        2023,
+        11,
+        21,
+        1,
+      );
+
+      const expectedComplexHumanText = "Tuesday, November 21, 2023 01:00";
+
+      // Use the service to get human time
+      final actualComplexHumanText = humanTimeService.textTimeAbsolute(
+        time2,
+      );
+
+      // Check that the specific absolut time value is correct
+      expect(
+        actualComplexHumanText,
+        expectedComplexHumanText,
+      );
+    });
+
+    test("Check special case 'now' relative time", () {
+      // Setup actual relative values for 'now' case
+      final nowTime = constantTestingTime.add(const Duration(seconds: 1));
```

Comment:
I think that we should substract the time here, because it's weird to check how much time passed since a date in the future.

Reply:
Yes, it does not matter, as the `timeago` library does check this and displays now instead. Also, pretty much all the default posts that we have in all tests will end up being in the future, as the base fixed date that I chose is in the year `2000` so in the past. In the end I could place the fixed date in the future to avoid this issue, but I does not change anything as at some point it will end up being in the past anyway so why not set it in the past directly :)

I still fixed your suggestion in c0be566

Reply:
Thanks for the clarifications!